src/coredump/coredump.c

```diff
@@ -1092,6 +1092,12 @@ static int process_socket(int fd) {
                         goto finish;
                 }
 
+                /* Only zero length messages are allowed after the first message that carried a file descriptor. */
+                if (!first && n > 0) {
+                        r = log_error_errno(SYNTHETIC_ERRNO(EBADMSG), "Received unexpected message with non zero length.");
+                        goto finish;
+                }
```

Comment:
well, you check n == 0 in the next line anyway, i.e. the reverse. hence, maybe move the above just below that next if block so that you can drop the n > 0 check

Reply:
and while you are at it, maybe get remove one level of indentation of the cmsg_close_all() after the big `if (n == 0) … else` below (i.e drop the "else"), after all the if block never allows fall through anyway.